api/v2.0/swagger.yaml

```diff
@@ -2744,6 +2744,39 @@ paths:
           $ref: '#/responses/409'
         '500':
           $ref: '#/responses/500'
+  /usergroups/search:
+    get:
+      summary: Search groups by groupname or ldapgroupdn
+      description: |
+        This endpoint is to search groups by group name or LDAP group DN.  It's open for all authenticated requests.
+      tags:
+        - usergroup
+      operationId: searchUserGroups
+      parameters:
+        - $ref: '#/parameters/requestId'
+        - $ref: '#/parameters/page'
+        - $ref: '#/parameters/pageSize'
+        - name: groupname
+          in: query
+          type: string
+          required: false
+          description: Group name for filtering results.
+        - name: ldapgroupdn
+          in: query
+          type: string
+          required: false
+          description: LDAP group DN for filtering results.  
+      responses:
+        '200':
+          description: Search groups successfully.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/UserGroup'
+        '401':
```

Comment:
403？

Reply:
all authenticated user could search user group, so there is no 403.